com.unity.ml-agents/Runtime/InplaceArray.cs

```diff
@@ -0,0 +1,250 @@
+using System;
+using System.Collections.Generic;
+
+namespace Unity.MLAgents
+{
+    /// <summary>
+    /// An array-like object that stores up to four elements.
+    /// This is a value type that does not allocate any additional memory.
+    /// </summary>
+    /// <remarks>
+    /// This does not implement any interfaces such as IList, in order to avoid any accidental boxing allocations.
+    /// </remarks>
+    /// <typeparam name="T"></typeparam>
+    public struct InplaceArray<T> where T : struct
```

Comment:
Is there any plans of adding methods for converting between this and barracuda tensorshape?

Reply:
Maybe we could do it as an extension method somewhere. Where would it be used?

Reply:
Let's save this for another PR - don't think it'll affect the public API